tools/destroy_virtual_hardware.sh

```diff
@@ -26,11 +26,38 @@ if [[ "$(id -u)" -ne 0 ]]; then
 fi
 
 function warn {
+    set +x
     echo -e "\e[1;31m$1\e[0m"
+    set -x
 }
 
 function success {
+    set +x
     echo -e "\e[1;36m$1\e[0m"
+    set -x
+}
+
+function verify_omicron_uninstalled {
+    svcs "svc:/system/illumos/sled-agent:default" 2>&1 > /dev/null
+    if [[ $? -eq 0 ]]; then
+        set +x
+        warn "Omicron is still installed, please run \`omicron-package uninstall\`, and then re-run this script"
+        exit 1
+    fi
+}
+
+function unload_xde_driver {
+    local ID="$(modinfo | grep xde | cut -d ' ' -f 1)"
+    if [[ "$ID" ]]; then
+        local RC=0
```

Comment:
I'm not really suggesting changing any of this, but more of an FYI: a common pattern I've seen is to have a `fail` or `fatal` function just like `warn` but that exits 1 afterwards.  Then L52-L58 can just become `modunload -i "$ID" || fail "Failed to unload xde driver"`.

Reply:
Fixed with added cleanup in [6e4fd84](https://github.com/oxidecomputer/omicron/pull/1352/commits/6e4fd84d2eaa47e8de9a4a241c65e5ea5391d386)